src/dune_rules/opam_create.ml

```diff
@@ -228,6 +228,10 @@ let opam_fields project (package : Package.t) =
   in
   let list_fields =
     [ "maintainer", Package_info.maintainers info
+    ; ( "x-maintenance-intent"
+      , match Package_info.maintenance_intent info with
+        | None -> Some [ "(latest)" ]
```

Comment:
While this is probably reasonable, I think this should be only generated if dune-lang is >= 3.18, otherwise upgrading to 3.18 will unconditionally rewrite people's OPAM files.

Reply:
Ha good point thank you! The opam files should indeed be stable and not change because of a dune update, I've gated the `x-maintenance-intent` to dune >= 3.18 as you suggested.